src/main/java/uk/gov/hmcts/darts/authentication/client/impl/OAuthClientImpl.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.hmcts.darts.authentication.client.impl;
+
+import com.nimbusds.oauth2.sdk.AuthorizationCode;
+import com.nimbusds.oauth2.sdk.AuthorizationCodeGrant;
+import com.nimbusds.oauth2.sdk.AuthorizationGrant;
+import com.nimbusds.oauth2.sdk.TokenRequest;
+import com.nimbusds.oauth2.sdk.auth.ClientAuthentication;
+import com.nimbusds.oauth2.sdk.auth.ClientSecretBasic;
+import com.nimbusds.oauth2.sdk.auth.Secret;
+import com.nimbusds.oauth2.sdk.http.HTTPResponse;
+import com.nimbusds.oauth2.sdk.id.ClientID;
+import lombok.SneakyThrows;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.darts.authentication.client.OAuthClient;
+import uk.gov.hmcts.darts.authentication.config.AuthProviderConfigurationProperties;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+
+@Component
+public class OAuthClientImpl implements OAuthClient {
```

Comment:
Replace the use of feign as:-

1) This class avoids our code being bound to a specific yaml key. 
2) The use of nimbus is consistent with the way in which we validate tokens.

Reply:
Concious that HMCTS standards have a preference for Feign (https://hmcts.github.io/standards/standards/java.html), so it may be better to keep our http clients consistent with that, unless we get any compelling benefits from using nimbus for this one use case?

Reply:
We can use Nimbus for JWT operations. Platops confirmed.